File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length() == 0) return 0;
        int i = 0 , j = 1;
        int size = s.length();
        Set<Character> seen = new HashSet<>();
        int ans = 1;
        seen.add(s.charAt(i));
        while ( j < size ) {
            char curr = s.charAt(j);
            if(seen.contains(curr)) {
                while(i < j) {
                    if(s.charAt(i) == curr) {
                        i++;
                        break;
                    } else {
                        seen.remove(s.charAt(i));
                        i++;                        
                    }
                }
            } else {
                ans = Math.max(ans, j - i + 1);
                seen.add(curr); 
            }
            j++;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset = set(nums)
        longest = 0

        for n in nums:
            if (n - 1) not in numset:
                curr = 1
                while curr + n in numset:
                    curr += 1
                longest = max(longest, curr)
        return longest